tidymodels-regression-part2.qmd

```diff
@@ -0,0 +1,274 @@
+# Regression in Tidymodels - Part 2
+
+This chapter is starts off as the Regression in Tidymodels chapter - Epidemiological time series edition and goes from there to making (slightly) more sophisticated predictions and interactive plots.
+
+```{r}
+#| echo: false
+source("_common.R")
+```
+
+## Libraries and data
+
+```{r, message = FALSE}
+# Load necessary packages
+library(tidymodels)
+library(plotly)
+```
+
+As a follow-up to the Regression in `Tidymodels` chapter, we're going to look at how the functions we used to perform linear regression fair when applied to a small set of epidemiological time series data. We'll be working with the built-in data on daily confirmed COVID-19 case and death rates for California over Dec. 31, 2020 to Dec. 31, 2021 (this was originally compiled by JHU).
+
+Note that while data is available for all states, we're focusing on just one state's worth of data to avoid having to manage multiple regions (panel data) just yet. 
+
+```{r}
+# Load and subset data
+ca_case_death_rate_subset <- case_death_rate_subset %>% 
+  filter(geo_value == "ca")
+```
+
+## Simple linear regression
+
+Our goal for this exercise is to construct a simple linear regression model of daily death rate as a function of daily case rate. And to do this, we'll essentially work through the same routine that we went through above (just in a different scenario). Thus, we'll opt for the same linear regression model specification. As we mentioned previously, setting the mode is unnecessary for linear regression, so let's drop that bit this time round. In addition, `lm` is the default engine for linear regression, so, we don't have to include that bit either.
+
+```{r, echo = FALSE, out.width = "50%", fig.align='center'}
+knitr::include_graphics("img/set_engine.png")
+```
+
+```{r}
+lm_spec_way2 <- linear_reg() 
+```
+
+Now we fit our model by inputting the formula (of the form `y ~ x`) `death_rate ~ case_rate` and the `ca_case_death_rate_subset` into the `fit()` function.
+
+```{r}
+ca_lm_fit <- lm_spec_way2 %>%
+  fit(death_rate ~ case_rate, data = ca_case_death_rate_subset)
+ca_lm_fit
+```
+
+As before, let's use `purrr`’s `pluck()` function to access the underlying fit and then pipe that into `summary()` to get a basic synopsis of our model.
+
+```{r}
+ca_lm_fit %>% 
+  pluck("fit") %>%
+  summary()
```

Comment:
Did you describe `pluck()` previously (grabs a list component)? 

In this context you could also do
```r
extract_fit_engine(ca_lm_fit) %>% summary()
```

Reply:
In the previous chapter on regression, I used the pluck function (& I added a bit of description based on what you said here).